forge/test/mlir/llama/test_llama_inference.py

```diff
@@ -126,3 +127,141 @@ def test_llama_inference_cache_cpu():
     # Generated text
     generated_text = tokenizer.decode(generated_tokens[0], skip_special_tokens=True)
     print(generated_text)
+
+
+@pytest.mark.xfail()
+def test_llama_inference_no_cache_tt():
+
+    # Load Llama 3B model and tokenizer
+    model_path = "openlm-research/open_llama_3b"
+    framework_model = load_model(model_path, use_cache=False)
+    tokenizer = LlamaTokenizer.from_pretrained(model_path)
+
+    # Prepare input sentence
+    prompt = "Q: What is the largest animal?\nA:"
+    inputs = tokenizer(prompt, return_tensors="pt")
+
+    # Run Prefill on CPU with no cache to get the initial logits
+    logits = framework_model(input_ids=inputs.input_ids, attention_mask=inputs.attention_mask)
+    next_token_logits = logits[0][:, -1, :]
+    next_token = torch.argmax(next_token_logits, dim=-1)
+
+    input_ids = torch.cat(
+        [inputs.input_ids, next_token.unsqueeze(0)], dim=-1
+    )
+
+
+    # Run decode stage on TT device and generate tokens by appending predicted token into sequence of input tokens
+    # untill the a specified maximum number of new tokens is reached or an end-of-sequence token is encountered.
+    max_new_tokens = 32
+    for _ in range(max_new_tokens):
+
+        # CPU Inference
+        framework_output = framework_model(input_ids)
+
+        # Compile the model
+        tt_inputs = [input_ids]
+        compiled_model = forge.compile(framework_model, sample_inputs=tt_inputs)
```

Comment:
Do we need to compile the model in the loop? Instead, let's try to use fixed max seq len and compile model only once :))

Reply:
Instead of compile the model in loop, create a input_ids with max seq len by applying padding and passing input with max seq len and attention mask (i.e to avoid performing attention on padding token indices).